core/services/ocr2/plugins/rebalancer/plugin.go

```diff
@@ -404,42 +407,36 @@ func (p *Plugin) computePendingTransfersConsensus(observations []models.Observat
 	return quorumEvents, nil
 }
 
-// computeMedianGraph computes a graph with the provided median liquidities per chain.
-// The nodes in the resulting graph are copied from the plugin node. Meaning that different
-// plugin instances might have different graph.
-func (p *Plugin) computeMedianGraph(medianLiquidities []models.NetworkLiquidity) (liquiditygraph.LiquidityGraph, error) {
-	g := liquiditygraph.NewGraph()
-
-	for _, medianLiq := range medianLiquidities {
-		sourceNetwork := medianLiq.Network
-
-		neighbors, exists := p.liquidityGraph.GetNeighbors(sourceNetwork)
-		if !exists {
-			p.lggr.Warnw("neighbors not found for network", "network", sourceNetwork)
-			continue
+func (p *Plugin) computeGraphLanesConsensus(observations []models.Observation) []models.Lane {
+	counts := make(map[models.Lane]int)
+	for _, obs := range observations {
+		for _, lane := range obs.Lanes {
+			counts[lane]++
 		}
+	}
 
-		switch g.HasNetwork(sourceNetwork) {
-		case true: // was already added to the graph, as a neighbor of another network that was processed first.
-			if !g.SetLiquidity(sourceNetwork, medianLiq.Liquidity) {
-				return nil, fmt.Errorf("graph set liquidity %s network %d", medianLiq.Liquidity, sourceNetwork)
-			}
-		case false: // seen for first time
-			if !g.AddNetwork(sourceNetwork, medianLiq.Liquidity) {
-				return nil, fmt.Errorf("graph add network liquidity %s network %d", medianLiq.Liquidity, sourceNetwork)
-			}
+	var quorumLanes []models.Lane
+	for lane, count := range counts {
+		if count >= p.f+1 {
+			quorumLanes = append(quorumLanes, lane)
 		}
+	}
 
-		for _, destNetwork := range neighbors {
-			if !g.HasNetwork(destNetwork) {
-				if !g.AddNetwork(destNetwork, big.NewInt(0)) {
-					return nil, fmt.Errorf("graph add dest network %d unexpectedly returned false", destNetwork)
-				}
-			}
+	return quorumLanes
+}
 
-			if err := g.AddConnection(sourceNetwork, destNetwork); err != nil {
-				return nil, fmt.Errorf("graph add connection %d->%d: %w", sourceNetwork, destNetwork, err)
-			}
+// computeMedianGraph computes a graph with the provided median liquidities per chain and lanes that quorum agreed on.
+func (p *Plugin) computeMedianGraph(
+	lanes []models.Lane, medianLiquidities []models.NetworkLiquidity) (liquiditygraph.LiquidityGraph, error) {
+
+	g, err := liquiditygraph.NewGraphFromLanes(lanes)
+	if err != nil {
+		return nil, fmt.Errorf("new graph from lanes: %w", err)
+	}
+
+	for _, medianLiq := range medianLiquidities {
+		if !g.SetLiquidity(medianLiq.Network, medianLiq.Liquidity) {
+			p.lggr.Infow("median liquidity on network not found on lanes quorum", "net", medianLiq.Network)
```

Comment:
Is this an error situation?

```suggestion
			p.lggr.Errorw("median liquidity on network not found on lanes quorum", "net", medianLiq.Network)
```

Reply:
Yeah maybe should be an error.
At that point we should expect that all nodes see the same observations and have the exact same graph.